task-sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -128,32 +124,26 @@
     TerminalTIState.SUCCESS,
 ]
 
-
-@overload
-def mkpipe() -> tuple[socket, socket]: ...
-
-
-@overload
-def mkpipe(remote_read: Literal[True]) -> tuple[socket, BinaryIO]: ...
+# Setting a fair buffer size here to handle most message sizes. Intention is to enforce a buffer size
+# that is big enough to handle small to medium messages while not enforcing hard latency issues
+BUFFER_SIZE = 4096
```

Comment:
Is it worth documenting it somewhere that message above 4kb will still fail?

Reply:
or do we want to make this configurable?

Reply:
Yeah, it's just a "lets be a little bit more performant without blowing up memory usage if it's not needed" It's such a low level thing that for now I'd say lets not make this configurable unless we have clearer data that this size actually affects anything.

In some regards this _might_ have been premature optimization to even do it.